server/issue.go

```diff
@@ -158,6 +158,159 @@ func httpAPIGetCreateIssueMetadata(ji Instance, w http.ResponseWriter, r *http.R
+			errors.WithMessage(err, "failed to get CreateIssue metadata")
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	b, err := json.Marshal(cimd)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to marshal response")
+	}
+	_, err = w.Write(b)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to write response")
+	}
+
+	return http.StatusOK, nil
+}
+
+func httpAPIAttachIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	if r.Method != http.MethodPost {
+		return http.StatusMethodNotAllowed,
+			errors.New("method " + r.Method + " is not allowed, must be POST")
+	}
+
+	api := ji.GetPlugin().API
+
+	comment := &struct {
+		PostId string       `json:"post_id"`
+		Fields jira.Comment `json:"fields"`
+	}{}
+	err := json.NewDecoder(r.Body).Decode(&comment)
+	if err != nil {
+		return http.StatusBadRequest,
+			errors.WithMessage(err, "failed to decode incoming request")
+	}
+
+	mattermostUserId := r.Header.Get("Mattermost-User-Id")
+	if mattermostUserId == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	jiraUser, err := ji.GetPlugin().LoadJIRAUser(ji, mattermostUserId)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	jiraClient, err := ji.GetJIRAClient(jiraUser)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	// Lets add a permalink to the post in the Jira Description
+	post, appErr := api.GetPost(comment.PostId)
+
+	if appErr != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(appErr, "failed to load post "+comment.PostId)
+	}
+
+	mattermostUser, err := api.GetUser(post.UserId)
+	if err != nil {
+		// TODO : appropriate error
+		// return http.StatusInternalServerError, err
+	}
+
+	if post == nil {
+		return http.StatusInternalServerError,
+			errors.New("failed to load post " + comment.PostId + ": not found")
+	}
+
+	if channel, _ := api.GetChannel(post.ChannelId); channel != nil {
+		if team, _ := api.GetTeam(channel.TeamId); team != nil {
+			permalink := fmt.Sprintf("%v/%v/pl/%v",
+				ji.GetPlugin().GetSiteURL(),
+				team.Name,
+				comment.PostId,
+			)
+
+			permalinkMessage := fmt.Sprintf("*@%s attached a* [message|%s] *from @%s*\n", jiraUser.User.Name, permalink, mattermostUser.Username)
+
+			if len(comment.Fields.Body) > 0 {
+				comment.Fields.Body += fmt.Sprintf("\n%v", permalinkMessage)
+			} else {
+				comment.Fields.Body = permalinkMessage
+			}
+
+			comment.Fields.Body += post.Message
+		}
+	}
+
+	commentAdded, _, err := jiraClient.Issue.AddComment(comment.Fields.ID, &comment.Fields)
+	fmt.Printf("commentAdded = %+v\n", commentAdded)
+
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to attach the comment, postId: "+comment.PostId)
+	}
+
+	// Reply to the post with the issue link that was created
+	reply := &model.Post{
+		// TODO: Why is this not created.Self?
```

Comment:
TODO? Not sure what you mean.

Reply:
@crspeller I think this may be a leftover from my comment in CreateIssue. We compose the issue URL, rather than using what I think is a "self" reference provided by Jira. Wanted to test out and maybe switch over.

Reply:
self returns a Jira Rest endpoint for the ticket, which provides quite a bit of information about the ticket, but no single property that contains the complete URL to the Jira issue.  Looks like we will have to construct the URL

Reply:
You can totally just kill the comment here for now